pkg/crds/yaml/generated/management.cattle.io_globalroles.yaml

```
@@ -51,6 +51,59 @@ spec:
             type: string
           metadata:
             type: object
+          namespacedRules:
+            additionalProperties:
+              items:
+                description: PolicyRule holds information that describes a policy
+                  rule, but does not contain information about who the rule applies
+                  to or which namespace the rule applies to.
+                properties:
+                  apiGroups:
+                    description: APIGroups is the name of the APIGroup that contains
+                      the resources.  If multiple API groups are specified, any action
+                      requested against one of the enumerated resources in any API
+                      group will be allowed. "" represents the core API group and
+                      "*" represents all API groups.
+                    items:
+                      type: string
+                    type: array
+                  nonResourceURLs:
+                    description: NonResourceURLs is a set of partial urls that a user
+                      should have access to.  *s are allowed, but only as the full,
+                      final step in the path Since non-resource URLs are not namespaced,
+                      this field is only applicable for ClusterRoles referenced from
+                      a ClusterRoleBinding. Rules can either apply to API resources
+                      (such as "pods" or "secrets") or non-resource URL paths (such
+                      as "/api"),  but not both.
+                    items:
+                      type: string
+                    type: array
+                  resourceNames:
+                    description: ResourceNames is an optional white list of names
+                      that the rule applies to.  An empty set means that everything
+                      is allowed.
+                    items:
+                      type: string
+                    type: array
+                  resources:
+                    description: Resources is a list of resources this rule applies
+                      to. '*' represents all resources.
+                    items:
+                      type: string
+                    type: array
+                  verbs:
+                    description: Verbs is a list of Verbs that apply to ALL the ResourceKinds
+                      contained in this rule. '*' represents all verbs.
+                    items:
+                      type: string
+                    type: array
+                required:
+                - verbs
+                type: object
+              type: array
+            description: NamespacedRules are the rules that are active in each namespace
+              of this GlobalRole. Does not support *.
```

Please expand on what "Does not support *" mean in this context, as the documentation picks up the underlying types (as per above) which makes it confusing to the end user. It would be nice if we called out any validation that blocks its use, or in case we don't enforce it, what would be the expected side-effects of using `*`.

---

Agree with what Paulo says here, I would change this to something like the following:
```suggestion
            description: NamespacedRules are the rules that are active in each namespace
              of this GlobalRole. * has no special meaning in the keys - these keys are read 
              as raw strings and must exactly match with one existing namespace. 
```

---

Wouldn't it be less confusing to say the key needs to be a valid namespace name (matching the following regex: `[a-z0-9]([-a-z0-9]*[a-z0-9])?`). In that case no need at all to mention `*` (which on first read confused me as to why that would be important to note).

---

I think that a regex in the comment would be more confusing. The key information to convey here is that there is no wildcard support and each key needs to exactly match (no pattern matching) a namespace - I'm not sure that a regex gets us closer to that.

---

I see how my message above might have been confusing, I was not suggesting we literally put the regex in the description. I was suggesting simply that this fields needs to be a valid namespace name. (Then the user can infer by themselves that there is no support for wildcard).

eg: "NamespacedRules are the rules that are active in each namespace of this GlobalRole. Each key must be in the format of a valid namespace name."

But I hadn't even considered that the user my think `foo-*` would work, so maybe it is a good idea to include that in the description as you suggest.